### PR TITLE
Update npm packages to resolve CH issues

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 ### Prerequisites
 
-1. [Node.js](https://nodejs.org/) 14.16
-1. npm 6.13.4
+1. [Node.js](https://nodejs.org/) 16.14.2
+1. npm 8.13.0
```
Comment (DonJayamanne) [inline CONTRIBUTING.md:9]: Looks like 8.13.0 was used when generating the pakcage lock in the past, that seems to be the best match.
8.3.0 makes a tonne of changes, and 8.13.0 has the least changes

Comment (DonJayamanne) [reply CONTRIBUTING.md:9]: Still need to test the build and the vsix

File: package.json [hunk 2]
```diff
@@ -125,16 +123,10 @@
         "re-resizable": "~6.5.5",
         "react": "^16.8.4",
         "react-dom": "^16.8.4",
-        "react-is": "^18.1.0",
-        "react-syntax-highlighter": "^15.5.0",
```
Comment (DonJayamanne) [inline package.json:129]: None of this was required.
Looks like we were trying to force the dependencies a numbre of ways by installing manually and also using resolutions.

File: package.json [hunk 3]
```diff
@@ -181,15 +172,13 @@
     "optionalDependencies": {
         "canvas": "^2.7.0"
     },
-    "resolutions": {
-        "trim": "0.0.3",
-        "tar": "4.4.18",
+    "overrides": {
         "ini": "1.3.6",
         "simple-get": "3.1.1",
-        "marked": "4.0.10",
```
Comment (DonJayamanne) [inline package.json:189]: `marked` has been updated in the child depdencies

File: package.json [hunk 4]
```diff
@@ -181,15 +172,13 @@
     "optionalDependencies": {
         "canvas": "^2.7.0"
     },
-    "resolutions": {
-        "trim": "0.0.3",
-        "tar": "4.4.18",
+    "overrides": {
```
Comment (DonJayamanne) [inline package.json:175]: Using built (new to npm 8) in mechanism to enforce versions

File: package.json [hunk 5]
```diff
@@ -181,15 +172,13 @@
     "optionalDependencies": {
         "canvas": "^2.7.0"
     },
-    "resolutions": {
-        "trim": "0.0.3",
```
Comment (DonJayamanne) [inline package.json:185]: These aren't required, doesn't seem to be used, hence I've removed them.

File: package.json [hunk 6]
```diff
@@ -125,16 +123,10 @@
         "re-resizable": "~6.5.5",
         "react": "^16.8.4",
         "react-dom": "^16.8.4",
-        "react-is": "^18.1.0",
-        "react-syntax-highlighter": "^15.5.0",
         "react-toggle": "^4.1.2",
-        "strip-ansi": "^5.2.0",
         "styled-components": "5.2.1",
-        "tar": "^4.4.18",
-        "trim": "^0.0.3",
```
Comment (DonJayamanne) [inline package.json:134]: Looks like we have them in resolutions as well as dependencies, remove them
Guess it was a way to force things to work

File: package.json [hunk 7]
```diff
@@ -164,13 +156,12 @@
...
         "node-loader": "^2.0.0",
-        "npm-force-resolutions": "0.0.10",
         "prettier": "^2.0.5",
         "style-loader": "^1.2.1",
         "svg-inline-loader": "^0.8.2",
         "thread-loader": "^3.0.4",
-        "ts-loader": "^7.0.5",
-        "typescript": "^3.9.3",
+        "ts-loader": "^9.3.1",
+        "typescript": "^4.7.4",
```
Comment (DonJayamanne) [inline package.json:164]: Had to update ts as well, due to new vscode dependencies (the *.d.ts files require later versions of TS)

File: package.json [hunk 8]
```diff
@@ -93,7 +93,6 @@
         "package": "vsce package -o ms-notebook-renderers.vsix",
         "test": "node ./out/test/runTest.js",
         "prettier-fix": "prettier 'src/**/*.ts*' --write && prettier 'build/**/*.js' --write",
-        "preinstall": "npx npm-force-resolutions",
```
Comment (DonJayamanne) [inline package.json:96]: Removed in favour of builtin `overrides`

**Review by amunger — APPROVED** [review]

**Review by amunger — APPROVED** [review]

**Pull request merged**